File: cliente-cpp_entrega_1/src/Produto.cpp
#include "../include/Produto.hpp"

Produto::Produto(){};

Produto::Produto(int i, const string& n, const double& p, const string& f)
    : id(i), nome(n), preco(p), fabricante(f){}


int Produto::getId() const{
    return this->id;
}

void Produto::setId(int i){
    this->id = i;
}

string Produto::getNome() const{
    return this->nome;
}

void Produto::setNome(const string& n){
    this->nome = n;
}

void Produto::setPreco(double p){
    this->preco = p;
}

double Produto::getPreco() const{
    return this->preco;
}

void Produto::setFabricante(const string& f){
    this->fabricante = f;
}

string Produto::getFabricante() const{
    return this->fabricante;
}

string Produto::toString() const{
    return "Produto{id=" + to_string(this->getId()) + 
    ", nome='" + this->getNome() + 
    "', preco=" + to_string(this->getPreco()) + 
    ", fabricante='" + this->getFabricante() + "'}";
}

File: rmi-sistema/cliente-java/src/entidade/Produto.java
package entidade;

/**
 * Produto — entidade base (1 de 4)
 * Superclasse de toda a hierarquia ("é-um").
 */
public class Produto {

    private int    id;
    private String nome;
    private double preco;
    private String fabricante;

    public Produto() {}

    public Produto(int id, String nome, double preco, String fabricante) {
        this.id         = id;
        this.nome       = nome;
        this.preco      = preco;
        this.fabricante = fabricante;
    }

    public int    getId()          { return id; }
    public void   setId(int id)    { this.id = id; }

    public String getNome()                { return nome; }
    public void   setNome(String nome)     { this.nome = nome; }

    public double getPreco()               { return preco; }
    public void   setPreco(double preco)   { this.preco = preco; }

    public String getFabricante()                  { return fabricante; }
    public void   setFabricante(String f)          { this.fabricante = f; }

    @Override
    public String toString() {
        return "Produto{id=" + id + ", nome='" + nome +
               "', preco=" + preco + ", fabricante='" + fabricante + "'}";
    }
}